util/exercise/src/cmd/configure.rs

```diff
@@ -53,19 +59,28 @@ fn get_user_config(exercise_name: &str, config_content: &Value) -> Result<Value>
 
             continue;
         } else {
-            break user_input;
+            break Some(user_input);
         };
     };
 
+    let core = unlocked_by.is_none();
+
     let difficulty = loop {
-        let unlocked_by = get!(config_content, "exercises", as_array)
-            .iter()
-            .find(|exercise| exercise["slug"] == unlocked_by)
-            .ok_or(format_err!(
-                "exercise '{}' not found in config",
-                unlocked_by
-            ))?;
-        let unlocked_by_difficulty = get!(unlocked_by, "difficulty", as_u64);
+        let unlocked_by_difficulty = match unlocked_by {
+            Some(ref unlocked_by) => {
+                let unlocked_by_exercise = get!(config_content, "exercises", as_array)
+                    .iter()
+                    .find(|exercise| exercise["slug"] == unlocked_by.as_str())
```

Comment:
We search for an exercise several times in this codebase. Do you think it might be worth extracting a function like 

```rust
fn find_exercise<F>(config_content: Value, finder: F) -> Option<Value> where F: FnOnce(Value) -> bool {
    get!(config_content, "exercises", as_array)
                    .iter()
                    .find(finder)
}
```

Then this expression, and similar ones like it elsewhere, could be compacted into one line:

```rust
let unlocked_by_exercise = find_exercise(config_content, |exercise| exercise["slug"] == unlocked_by.as_str())
```

Reply:
I tried it and it added more code than it saved, so I'm reverting this suggestion.

Reply:
Just tried it myself - same result